control_flow.py

```diff
@@ -0,0 +1,338 @@
+
+    def reset_dispatch_record(self):
+        self.current_dispatching_op = None
+        self.already_dispatched_keys = None
+
+    def record_dispatch(self, dispatch_key, operator):
+        self.currently_dispatching_op = operator
+        if self.already_dispatched_keys is None:
+            self.already_dispatched_keys = DispatchKeySet(dispatch_key)
+        else:
+            self.already_dispatched_keys = self.already_dispatched_keys | DispatchKeySet(dispatch_key)
+
+
+dispatcher_singleton = PyDispatcher()
+
+class PyOperator:
+    def __init__(self, name):
+        self.name = name
+        self.table = {}
+
+        # TODO: torch_dispatch expects PyOperator to be an instance of a torch.ops.aten op.
+        self.overloadpacket = self
+
+        # Hack for FX tracing
+        self.__name__ = f'torch.{name}'
+
+    def impl(self, dispatch_key, fn):
+        assert dispatch_key not in self.table
+        self.table[dispatch_key] = fn
+
+    def fallthrough(self, dispatch_key):
+        assert dispatch_key not in self.table
+        self.table[dispatch_key] = fallthrough_fn(self, dispatch_key)
+
+    def __call__(self, *args, **kwargs):
+        flat_args = to_flat_tuple(args, kwargs)
+        if has_torch_function(flat_args):
+            return handle_torch_function(self, flat_args, *args, **kwargs)
+
+        return dispatcher_singleton.call(self, args, kwargs)
+
+
+def compute_dispatch_key(PyOperator, args, kwargs, additional_exclude=None):
+    tensors = get_tensors(args, kwargs)
+    dispatch_key = key_extractor(tensors, additional_exclude)
+    return dispatch_key
+
+
+def dispatch(dispatch_key, operator, args, kwargs):
+    print("Dispatching:", dispatch_key, operator.__name__)
+    if dispatch_key not in SUPPORTED_KEYS:
+        raise RuntimeError(f'NYI: {dispatch_key}')
+    assert dispatch_key in operator.table
+    kernel = operator.table[dispatch_key]
+    return kernel(*args, **kwargs)
+
+
+def key_extractor(tensors, additional_exclude=None):
+    key_set = _C._dispatch_tls_local_include_set()
+    for tensor in tensors:
+        key_set = key_set | _C._dispatch_keys(tensor)
+    key_set = key_set - _C._dispatch_tls_local_exclude_set()
+    if additional_exclude is not None:
+        key_set = key_set - additional_exclude
+    return key_set.highestPriorityTypeId()
+
+
+def to_flat_tuple(args, kwargs):
+    flat_args, _ = tree_flatten(args)
+    flat_kwargs, _ = tree_flatten(kwargs)
+    flat_all = flat_args + flat_kwargs
+    return flat_all
+
+def get_tensors(args, kwargs):
+    flat_all = to_flat_tuple(args, kwargs)
+    tensor_args = [t for t in flat_all if isinstance(t, torch.Tensor)]
+    return tuple(tensor_args)
+
+"""
+We're going to define a `cond` operation.
+In order to do this, we need implementations for each of the dispatch keys.
+"""
+
+def cond_dense(pred, true_fn, false_fn, *operands):
+    print("Running cond dense", pred)
+    # print("Pred?", pred.code)
+    # if pred(tuple()):
+    if pred.elem:
```

Comment:
After applying the other change with extracting proxies this needs to be changed.
```suggestion
    if getattr(pred, "elem", False):
```